Footer updates (#1458)
carbon-design-system/carbon-for-ibm-dotcom-website: src/pages/components/footer.mdx

## Overview

> The Footer is a required component on all IBM.com pages. There are three sizes available: [Default](#default), [Short](#short) and [Micro](#micro), and each size has a specific use case. The Footer includes the IBM logo, navigational links, Locale or Language selector and required legal links.

Reviewer: ```suggestion
The Footer is a required component on all IBM.com pages. There are three sizes available: [default](#default), [short](#short) and [micro](#micro), and each size has a specific use case. The Footer includes the IBM logo, navigational links, locale or language selector and required legal links.
```

Suggest using exact prop/attribute cases to avoid confusion when they're used.

The Footer is generated automatically based on the country and language code that is set on the page in the metadata.

![Image of default footer](../../images/component/footer/footerV2-default-01.jpg)

<Caption>
  Default footer at 1584px max breakpoint, without optional adjunct legal links
</Caption>

## Default

> The Default footer includes all high-level corporate navigation, required legal links, locale selector, branding, and optional adjunct legal links.

Reviewer: ```suggestion
The default footer includes all high-level corporate navigation, required legal links, locale selector, branding, and optional adjunct legal links.
```

#### When to use

Default footer is suitable for most of the IBM.com pages. Examples include the IBM.com US home page.

### Anatomy

![Image of minimal footer](../../images/component/footer/footerV2-default-anatomy.jpg)

<Caption>
  Default footer at the 1584px (max) breakpoint with one adjunct legal link
</Caption>

1. **IBM logo:** Links to the IBM homepage and is a required branding element of the Footer. The IBM logo should not be altered.
1. **Link group heading:** The link group heading is present for all the link groups.

> 1. **Link:** The Default footer links are carefuly considered from the main navigation level and should not be changed.

Reviewer: ```suggestion
1. **Link:** The default footer links are carefully considered from the main navigation level and should not be changed.
```

Author: will fix carefully in another commit, leaving Default capitalized

1. **Locale selector:** Use the locale selector to trigger the locale modal component to allow users to change geographic regions and languages, if available.
1. **Legal links:** The general legal links are present on every IBM Footer and cannot be changed.
1. **Adjunct legal link(s) (Optional)**: Use up to two adjunct legal links for notes or messages legally required to provide regarding changes, policies or trademark issues.

### Behavior

Default footer spans the full width of the viewport and is the last element in the browser window. The Footer elements are persistent throughout the online experience. It is fully responsive and changes how elements are displayed based on the browser size.

#### Max breakpoint

> If the browser is wider than the max breakpoint of 1584px, the Default footer and all other page content will center and extra margin will appear on either side of the page layout.

Reviewer: ```suggestion
If the browser is wider than the max breakpoint of 1584px, the default footer and all other page content will center and extra margin will appear on either side of the page layout.
```

> #### Md and sm breakpoints

Reviewer: When referring to text that is used in code, should we use the exact case and wrap in code brackets, i.e. `md` and `sm`. Otherwise maybe we spell it out (Medium and small)? Just want to avoid any confusion when they are used.

> At the medium breakpoint, the Locale selector moves above the link groups and below the IBM logo. Additionally, at the small breakpoint, the link groups are nested into accordions. See the image below.

Reviewer: ```suggestion
At the medium breakpoint, the locale selector moves above the link groups and below the IBM logo. Additionally, at the small breakpoint, the link groups are nested into accordions. See the image below.
```

<Caption>
  From left to right: Default footer at 672px (md) and 320px (sm) breakpoints.
</Caption>

## Short

Short footer removes all the high-level corporate navigation, leaving the required legal links, locale selector, branding, and optional adjunct legal links.

#### When to use

Short footer is reserved for approved transactional pages, such as check out and registration experiences.

### Anatomy

![Image of minimal footer](../../images/component/footer/footerV2-short-anatomy.jpg)

<Caption>
  Short footer at the 1584px (max) breakpoint with one adjunct legal link.
</Caption>

1. **IBM logo:** Links to the IBM homepage and is a required branding element of the Footer. The IBM logo should not be altered.
2. **Locale selector:** Use the locale selector to trigger the locale modal component to allow users to change geographic regions and languages, if available.
3. **Legal links:** The general legal links are present on every IBM Footer and cannot be changed.
4. **Adjunct legal link(s) (optional)**: Use up to two adjunct legal links for notes or messages legally required to provide regarding changes, policies or trademark issues.

### Behavior

> The Short footer spans the full width of the viewport and is the last element in the browser window. It is fully responsive and changes how elements are displayed based on the browser size.

Reviewer: ```suggestion
The short footer spans the full width of the viewport and is the last element in the browser window. It is fully responsive and changes how elements are displayed based on the browser size.
```

#### Max breakpoint

> If the browser is wider than the max breakpoint of 1584px, the Short footer and all other page content will center and extra margin will appear on either side of the page layout.

Reviewer: ```suggestion
If the browser is wider than the max breakpoint of 1584px, the short footer and all other page content will center and extra margin will appear on either side of the page layout.
```

#### Md and sm breakpoints

At the medium breakpoint, the legal links stack into two columns.
At the small breakpoint, the legal links are stacked on top of each other in one column. See the image below.

![Image of Default footer](../../images/component/footer/footerV2-short-behaviour.jpg)

<Caption>
  From left to right: Short footer at 672px (md) and 320px (sm) breakpoints.
</Caption>

## Micro

> The Micro footer variation only shows the required legal links and the optional locale selector to maximize the space for the page content. Adjunct legal links are not allowed. If you need them, please use either the Default, or the Short footer.

Reviewer: ```suggestion
The micro footer variation only shows the required legal links and the optional locale selector to maximize the space for the page content. Adjunct legal links are not allowed. If you need them, please use either the default, or the short footer.
```

#### When to use

This Footer is reserved for stand-alone experiences such as product demos or slide shows. All other pages and experiences on IBM.com should use the Default or the Short variations.

### Anatomy

![Image of minimal footer](../../images/component/footer/footerV2-micro-anatomy.jpg)

<Caption>Micro footer at the 1584px (max) breakpoint</Caption>

1. **Legal links:** The general legal links are present on every IBM Footer and cannot be changed.
2. **Locale selector:** Use the locale selector to trigger the locale modal component to allow users to change geographic regions and languages, if available.

### Behavior

> The Micro footer spans the full width of the viewport and is the last element in the browser window. It is fully responsive and changes how elements are displayed based on the browser size.

Reviewer: ```suggestion
The micro footer spans the full width of the viewport and is the last element in the browser window. It is fully responsive and changes how elements are displayed based on the browser size.
```

#### Max breakpoint

If the browser is wider than the max breakpoint of 1584px, the Footer and all other page content will center and extra margin will appear on either side of the page layout.

#### Md and sm breakpoints

> At the medium breakpoint, the Micro footer will also stack the legal links. For the small breakpoint, the legal links are stacked on top of each other in one column.

Reviewer: ```suggestion
At the medium breakpoint, the micro footer will also stack the legal links. For the small breakpoint, the legal links are stacked on top of each other in one column.
```

![Image of Default footer](../../images/component/footer/footerV2-micro-behaviour.jpg)

<Caption>Micro footer at the 672px (md) and the 320px (sm) breakpoints</Caption>

## Modifiers

### Theme

As a mandatory component of the IBM.com web experience, the Footer should remain consistent, creating a branded anchor across all pages. The Footer is only available in the gray 90 theme to maintain consistency across all pages.

### Locale or Language selector

> All Footer sizes have the option to choose between using the Locale selector or Language selector.

Reviewer: ```suggestion
All Footer sizes have the option to choose between using the locale selector or language selector.
```

#### Locale selector

> The Footer defaults to the Locale selector, which displays the country and language of a given page within a button. It allows the user to change their locale through the [Locale modal](https://www.ibm.com/standards/carbon/components/localemodal) by selecting the desired region, followed by the desired language.

Reviewer: ```suggestion
The Footer defaults to the locale selector, which displays the country and language of a given page within a button. It allows the user to change their locale through the [locale modal](https://www.ibm.com/standards/carbon/components/localemodal) by selecting the desired region, followed by the desired language.
```

<Row>

<Column colMd={4} colLg={6}>

![A visual example of the Locale selector](../../images/component/footer/footer--modifier-locale.png)

> <Caption>The Locale selector, displaying United States – English.</Caption>

Reviewer: ```suggestion
<Caption>The locale selector, displaying United States – English.</Caption>
```

</Column>

</Row>

#### Language selector

The [Language selector](https://www.ibm.com/standards/carbon/components/language-selector) displays only the Language of a given page. It uses a combo box, and allows the user to select a language from the list or type in their desired language.

<Row>

<Column colMd={4} colLg={6}>

![A visual example of the Locale selector](../../images/component/footer/footer--modifier-language.png)

> <Caption>The Language selector, displaying English.</Caption>

Reviewer: ```suggestion
<Caption>The language selector, displaying English.</Caption>
```

</Column>

</Row>

### Adjunct legal links

> Only the Default and the Short footers may contain up to two adjunct legal links for legally required notes or messages. They are displayed at the very bottom of the page, under the IBM required legal links.

Reviewer: ```suggestion
Only the default and the short footers may contain up to two adjunct legal links for legally required notes or messages. They are displayed at the very bottom of the page, under the IBM required legal links.
```

![Image of micro footer](../../images/component/footer/footerV2-default-withAdjunctlegallinks.jpg)

<Caption>
  Default footer at the 1584px (max) breakpoint with one optional adjunct legal
  link
</Caption>

1. **Legal links:** The general legal links are present on every IBM Footer and cannot be changed.
2. **Adjunct legal link:** Use up to two adjunct legal links regarding changes, policies or trademark issues.

![Image of micro footer](../../images/component/footer/footerV2-short-withAdjunctlegallinks.jpg)

<Caption>
  Short footer at the 1312px (xlg) breakpoint, with one optional adjunct legal
  link
</Caption>

#### When to use

Use up to two adjunct legal links for legally required notes or messages legally required to provide regarding changes, policies or trademark issues.

> An adjunct legal link will open a web page or a modale window with detailed changes to policies or trademark issues.

Reviewer: ```suggestion
An adjunct legal link will open a web page or a modal window with detailed changes to policies or trademark issues.
```

#### When not to use

Do not use an adjunct legal link to write footnotes or content-related notices. Include only links with basic text; do not include imagery or special formatting.

## Content guidance

The links within the Footer cannot be changed, please contact the IA guild for any questions or request to alter the links.

| Element                                                                                                                                                                   | Content type | Required | Instances | Character limit <br/>(English / translated) | Notes                                                                                                        |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ | -------- | --------- | ------------------------------------------- | ------------------------------------------------------------------------------------------------------------ |

> | Logo                                                                                                                                                                      | Image        | Yes      | 1         | –                                           | Required, should not be altered, and links to IBM homepage. The logo is not required in the Micro variation. |

Reviewer: ```suggestion
| Logo                                                                                                                                                                      | Image        | Yes      | 1         | –                                           | Required, should not be altered, and links to IBM homepage. The logo is not required in the micro variation. |
```

> | Size                                                                                                                                                                      | Option list  | Yes      | 1         | –                                           | Default, Short, or Micro.                                                                                    |

Reviewer: ```suggestion
| Size                                                                                                                                                                      | Option list  | Yes      | 1         | –                                           | default, short, or micro.                                                                                    |
```

1. **Legal links:** The general legal links are present on every IBM footer and cannot be changed.
2. **Locale selector:** Use the locale selector to trigger the locale modal component to allow users to change geographic regions and languages, if available.

### Behavior

The micro footer spans the full width of the viewport and is the last element in the browser window. It is fully responsive and changes how elements are displayed based on the browser size.

#### Max breakpoint

If the browser is wider than the max breakpoint of 1584px, the footer and all other page content will center and extra margin will appear on either side of the page layout.

#### Md and sm breakpoints

> At medium breakpoint, the micro footer will also stack the legal links. For the small breakpoint, the legal links are stacked on top of each other on one column.

Reviewer 2: ```suggestion
At the medium breakpoint, the Micro Footer will also stack the legal links. For the small breakpoint, the legal links are stacked on top of each other in one column.
```

All footer sizes have the option to choose between using the locale selector or language selector.

#### Locale selector

The footer defaults to the locale selector, which displays the country and language of a given page within a button. It allows the user to change their locale through the [locale modal](https://www.ibm.com/standards/carbon/components/localemodal) by selecting the desired region, followed by the desired language.

<Row>

<Column colMd={4} colLg={6}>

![A visual example of the Locale selector](../../images/component/footer/footer--modifier-locale.png)

<Caption>The locale selector, displaying United States – English.</Caption>

</Column>

</Row>

#### Language selector

The [language selector](https://www.ibm.com/standards/carbon/components/language-selector) displays only the language of a given page. It uses a combo box, and allows the user to select a language from the list or type in their desired language.

<Row>

<Column colMd={4} colLg={6}>

![A visual example of the Locale selector](../../images/component/footer/footer--modifier-language.png)

<Caption>The language selector, displaying English.</Caption>

</Column>

</Row>

### Adjunct legal links

> Only the default and the short footers may contain up to 2 adjunct legal links for legally required notes or messages. They are displayed at the very bottom of the page, under the main legal links.

Reviewer 2: ```suggestion
Only the default and the short footers may contain up to two adjunct legal links for legally required notes or messages. They are displayed at the very bottom of the page, under the IBM required legal links.
```

![Image of micro footer](../../images/component/footer/footerV2-default-withAdjunctlegallinks.jpg)

<Caption>
  Default footer at the 1584px (max) breakpoint with one optional adjunct legal
  link
</Caption>

1. **Legal links:** The general legal links are present on every IBM footer and cannot be changed.
2. **Adjunct legal link:** Use up to two adjunct legal links regarding changes, policies or trademark issues.

![Image of micro footer](../../images/component/footer/footerV2-short-withAdjunctlegallinks.jpg)

<Caption>
  Short footer at the 1312px (xlg) breakpoint, with one optional adjunct legal
  link
</Caption>

#### When to use

> Use up to two adjunct legal links for legally required notes or messages legally required to provide, regarding changes, policies or trademark issues.

Reviewer 2: ```suggestion
Use up to two adjunct legal links for legally required notes or messages legally required to provide regarding changes, policies or trademark issues.
```

An adjunct legal link will open a web page or a modale window with detailed changes to policies or trademark issues.

#### When not to use

Do not use an adjunct legal link to write footnotes or content-related notices. Include only links with basic text; do not include imagery or special formatting.

## Content guidance

The links within the footer cannot be changed, please contact the IA guild for any questions or request to alter the links.

| Element                                                                                                                                                                   | Content type | Required | Instances | Character limit <br/>(English / translated) | Notes                                                                                                    |
| ------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------ | -------- | --------- | ------------------------------------------- | -------------------------------------------------------------------------------------------------------- |

> | Logo                                                                                                                                                                      | Image        | Yes      | 1         | –                                           | Required, should not be altered, links to IBM homepage. The logo is not required in the mirco variation. |

Reviewer 2: ```suggestion
| Logo                                                                                                                                                                      | Image        | Yes      | 1         | –                                           | Required, should not be altered, and links to IBM homepage. The logo is not required in the Micro variation. |
```
